node.js

```diff
@@ -49,7 +49,42 @@ module.exports = {
     'space-before-function-paren': off,
     'keyword-spacing': error,
     'eol-last': [ error, always ],
-    'brace-style': [ error, '1tbs', { allowSingleLine: true } ],
-    'curly': [ error, all ]
+    'brace-style': [ error, '1tbs', { allowSingleLine: false } ],
+    'curly': [ error, all ],
+    'block-spacing': [ error,  always ],
+    'indent': [ error, 2, { SwitchCase: 1 } ],
+    'no-tabs': error,
+    'spaced-comment': [ error, always ],
```

Comment:
Can we please just ban comments entirely? Comments are an excuse for bad programming.

Reply:
@antony I use them to delineate sections of code. For example, my saga files usually contain a `functions` section, a `saga` section and a `watchers` section. Similarly, in reducers, I separate the `actions` from the `action creators` and the `action creators` from the actual reducer via a comment. 

My personal opinion is we should allow comments, but if they get banned, I can change my code.

Reply:
You still need comments to explain magic numbers and business rules, right ?

Reply:
@ikhemissi there's a rule that I intend on proposing soon which is [`no-magic-numbers`](https://eslint.org/docs/4.0.0/rules/no-magic-numbers), the variable name should mean you can avoid a comment. That said, I highly doubt we're going to be banning comments any time soon

Reply:
I'd prefer good variable and function naming over comments, but I don't think we should ban comments completely. e.g. comments about potential code improvements that cannot be done at the current stage. I'd prefer comments over creating a JIRA 😄

Reply:
@ikhemissi use a good, verbose variable name to explain a magic number. Business rules should be expressed with clean, modular, well written code.

I realise I have extreme views on comments, but I detest them.

@jchen86 comments are litter, especially todos. There is no way of tracking how much work is required on a project, or how much tech debt there is if the work is all buried in the code to be discovered 4 years later. As for creating a JIRA, sure, JIRA is something I like even less than comments, but improvements to code should be either:
1. Listed with the other parts of the work in Trello, or whatever story tracking tool you use. They can be prioritised, triaged, and fitted into the delivery schedule.
2. Done, immediately, as they are obviously critical.
3. Forgotten, as they are obviously not so critical they need to be done right away, and they will likely end up in a comment, 4 years later, which will be completely irrelevant by the time anybody runs into it again, and even then, why is it so important for them to do it, when it wasn't so important for you!

So yeah. No comments > *

Reply:
@mayk93 If you have to deliniate sections of code, use different files and create an actual, physical boundary.

Comments usually indicate something else which isn't being done in the most optimal way.

Reply:
(btw I'm not suggesting we ban comments. It is my preference, but it's easy to see that I am in the minority here)

Reply:
> use a good, verbose variable name to explain a magic number. Business rules should be expressed with clean, modular, well written code.
@antony  I agree with you, we definitely must do this.
But I don't think good variable/function names will be enough to explain some quirks (that may or may not come from business or technical constraints): you will still need comments to do it
You may be able to do that in Git commit messages too but what you want after all is to be clear enough for other developers and for the future you to be able to pick up work fast and understand why a piece of code was written like that (if you ever need to do that, maybe you will never need to).
Basically, what I am saying is not to comment every line you write, but to not disallow comments as they may come in handy (plus I don't want my build to fail because there is a comment in there).